docs/application-development-guide.md

```diff
@@ -2904,6 +3294,30 @@ Content is not available
 
 Content is not available
 
+</TabItem>
+<TabItem value="python">
+
+To define a Signal, set the Signal decorator `@workflow.signal` on the Signal function inside your Workflow.
+
+```python
+@workflow.signal
+ async def complete_with_greeting(self) -> None:
+     self._complete.set()
+```
+
+The `@workflow.signal` decorator defines a method as a Signal. Signals can be asynchronous or synchronous functions at any hierarchy depth; however, if a method is override, the override must also be decorated.
+The method's arguments are also the Signal's arguments.
+You can have a name parameter to customize the Signal's name, otherwise it defaults to the unqualified method name.
+You can use `dynamic=True`, which means all other unhandled Signals fall through to this.
```

Comment:
Need to clarify that the "name parameter" and "dynamic=True" apply to the `@workflow.signal` decorator. Otherwise, reads like they are method parameters or something.